test/libevmasm/Optimiser.cpp

```diff
@@ -111,8 +112,18 @@ namespace
 			iter = eliminator.feedItems(iter, _input.end(), usesMSize);
 			bool shouldReplace = false;
 			AssemblyItems optimisedChunk;
+			auto runGas = [&](AssemblyItems const& items) {
+				GasMeter gasMeter{std::make_shared<KnownState>(eliminator.getKnownState()), solidity::test::CommonOptions::get().evmVersion()};
+				GasMeter::GasConsumption gas;
+				for (auto const& item: items)
+					gas += gasMeter.estimateMax(item);
+				return gas;
+			};
 			optimisedChunk = eliminator.getOptimizedItems();
-			shouldReplace = (optimisedChunk.size() < static_cast<size_t>(iter - orig));
+			shouldReplace = (
+				optimisedChunk.size()  < static_cast<size_t>(iter - orig) &&
+				!(runGas(AssemblyItems(orig, iter)) < runGas(optimisedChunk))
+			);
```

Comment:
This is not quite the same as actually happens in Assembly.cpp now, is it :-)?

Reply:
I think we should extract this bit in `Assembly.cpp` into a helper and use the helper here. Otherwise we're not even testing the changes made by the PR.

Reply:
Stop suggesting to refactor things. This is an intentional copy of `Assembly.cpp`, as explicitly stated above - the fact that this is weird, doesn't mean we should start taking half the codebase apart refactoring things randomly. The PR is a minor fix in something that's generally obsolete, to prevent it from doing more harm than good before it's finally removed - it's literally a waste of time to do this kind of refactoring work on it.

Reply:
> This is an intentional copy of `Assembly.cpp`

Except that it no longer is, you just said that yourself above.

I'm not suggesting a refactor for the sake of a refactor. I'm saying that due to this being a copy we're not testing the actual implementation here and IMO it's a problem if we're interested in it being correct. I mean, if the code is, as you say, obsolete then maybe we're not and maybe just recopying it is enough, but I don't think the suggestion was that outrageous. It shouldn't have been a copy to begin with.

Reply:
Does this PR introduce a new copy of something that wasn't there before? Was the scope and intention of this PR fixing and refactoring the testing setup? It was neither. Hence what you're suggesting is extending the scope of the PR to something that's an entirely independent issue - and one at that, that we'd never individually have solved. That's the very definition of a distraction and wasting time. Not too outrageous in isolation, but this is a pattern that results in overall throughput being near zero.